### PR TITLE
Integrating channel api

### PR DESCRIPTION
**Pull request by ayushdevelop**
# Related Issue

- https://github.com/code-mancers/g-bot/issues/22

# functionalities Added or Changed

- integrated the api for fetching channels with frontend

### CODE REVIEW

File: client/src/components/Standups/StandupCreate.tsx [hunk 1]
```diff
@@ -4,12 +4,32 @@ import {
...
+
+        setChannelList(channels);
+      } catch (error) {
+        throw error;
+      }
+    };
+
+    fetchChannelList();
+  }, []);
+  console.log(channelList);
```
Comment (SujayPrabhu96) [inline client/src/components/Standups/StandupCreate.tsx:32]: remove

File: client/src/components/Standups/StandupCreate.tsx [hunk 2]
```diff
@@ -4,12 +4,32 @@ import {
...
+import { useState, useEffect } from "react";
 
 const StandupCreate = () => {
+  const [channelList, setChannelList] = useState([]);
+
+  useEffect(() => {
+    const fetchChannelList = async () => {
+      try {
+        const data = await fetchChannels();
+
```
Comment (SujayPrabhu96) [inline client/src/components/Standups/StandupCreate.tsx:21]: remove empty line

File: client/src/apis/fetchChannels.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import instance from ".";
+
+export default async function fetchChannels() {
+  const response = await instance.get("/channels");
+
```
Comment (SujayPrabhu96) [inline client/src/apis/fetchChannels.ts:5]: remove empty line

File: client/src/apis/fetchChannels.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+  return response;
```
Comment (SujayPrabhu96) [inline client/src/apis/fetchChannels.ts:6]:
```suggestion
  return response.data;
```

File: client/src/components/Standups/StandupCreate.tsx [hunk 5]
```diff
@@ -4,12 +4,32 @@ import {
...
+        const channels = await data.data.data;
+
```
Comment (SujayPrabhu96) [inline client/src/components/Standups/StandupCreate.tsx:23]: remove empty line

File: client/src/components/Standups/StandupCreate.tsx [hunk 6]
```diff
@@ -19,14 +39,7 @@ const StandupCreate = () => {
...
         </ArrayInput>
-        <SelectInput
-          source="Channels"
-          choices={[
-            { id: "channel 1", name: "Channel 1" },
-            { id: "channel 2", name: "Channel 2" },
-            { id: "channel 3", name: "Channel 3" },
-          ]}
-        />
+        <Dropdown list={channelList} />
```
Comment (SujayPrabhu96) [inline client/src/components/Standups/StandupCreate.tsx:42]: As we are not writing test cases, do we need a component ? can we do
```suggestion
        <SelectInput choices={channelList} />
```

File: server/src/services/__mock__/channel.mock.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
+const mockChannelData = [
```
Comment (SujayPrabhu96) [inline server/src/services/__mock__/channel.mock.ts:1]: remove mock file

File: client/.env [hunk 8]
```diff
@@ -0,0 +1 @@
+REACT_APP_PUBLIC_URL = http://localhost:8000
```
Comment (SujayPrabhu96) [inline client/.env:1]: `.env` should not be pushed to repo. Rename it to `.env.example`

File: client/jest.config.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+module.exports = {
```
Comment (SujayPrabhu96) [inline client/jest.config.js:1]: remove this file

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

File: client/src/apis/fetchChannels.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import instance from ".";
```
Comment (SujayPrabhu96) [inline client/src/apis/fetchChannels.ts:1]: rename the file to `channels.ts`

File: client/src/apis/fetchChannels.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+export default async function fetchChannels() {
+  const response = await instance.get("/channels");
```
Comment (SujayPrabhu96) [inline client/src/apis/fetchChannels.ts:4]: Lets update our endpoint to `/api/v1/channels`. Please do the necessary changes in backend as well

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

File: client/.env (hunk 8 above)
Comment (ayushdevelop) [reply client/.env:1]: sorry sorry I did add it to .gitIgnore I think I forgot saving. Will fix it

File: client/src/components/Standups/StandupCreate.tsx [hunk 12]
```diff
@@ -4,12 +4,30 @@ import {
...
+
+import { useState, useEffect } from "react";
 
 const StandupCreate = () => {
+  const [channelList, setChannelList] = useState([]);
+
+  useEffect(() => {
+    const fetchChannelList = async () => {
+      try {
+        const data = await fetchChannels();
```
Comment (SujayPrabhu96) [inline client/src/components/Standups/StandupCreate.tsx:21]:
```suggestion
        const { data: channels } = await fetchChannels();
```

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

**Review by SujayPrabhu96 — APPROVED** [review]

**Pull request merged**